eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/configuration/batch/jobs/ClusteringVariantJobConfigurationTest.java

```diff
@@ -156,6 +163,17 @@ public void jobFromMongo() throws Exception {
         assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());
     }
 
+    @Test
+    @DirtiesContext
+    @UsingDataSet(locations = {"/test-data/submittedVariantEntityStudyReader.json"})
+    public void studyJobFromMongo() throws Exception {
+        JobExecution jobExecution = jobLauncherTestUtilsStudyFromMongo.launchJob();
+        List<String> expectedSteps = new ArrayList<>();
+        expectedSteps.add(STUDY_CLUSTERING_STEP);
+        assertStepsExecuted(expectedSteps, jobExecution);
+        assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());
+    }
+
```

Comment:
**Optional:** Similar to other jobs in the module, suggest adding tests for idempotency.